### DISCUSSION TITLE
SixLabors ImageSharp license was changed to include commercial restrictions on Jun 28, 2022

### DISCUSSION THREAD

**Discussion by FrankRay78** [spectreconsole/spectre.console#1072] [General]
On Jun 7, 2020 commit 874115bd0a382a922ddfea5dd33fe67ecef245f6 (https://github.com/SixLabors/ImageSharp/commit/874115bd0a382a922ddfea5dd33fe67ecef245f6) changed the license that ImageSharp was distributed under from GNU AFFERO GENERAL PUBLIC LICENSE (Version 3, 19 November 2007) to Apache License (Version 2.0, January 2004), see: https://github.com/SixLabors/ImageSharp/blob/874115bd0a382a922ddfea5dd33fe67ecef245f6/LICENSE

On Jun 28, 2022 commit cb3d073fa6ffaeada0c6ba35745542d225c2d541 (https://github.com/SixLabors/ImageSharp/blob/cb3d073fa6ffaeada0c6ba35745542d225c2d541/LICENSE) changed the license that ImageSharp was distributed under from Apache License (Version 2.0, January 2004) to Six Labors Split License (Version 1.0, June 2022) **which includes commercial restrictions**, see: https://github.com/SixLabors/ImageSharp/blob/cb3d073fa6ffaeada0c6ba35745542d225c2d541/LICENSE

This most recent license seems quite a departure from the previous Apache License and includes a number of restrictions, including:

```
   Works in Source or Object form are licensed to You under the Apache License, Version 2.0 if.

   - You are consuming the Work in for use in software licensed under an Open Source or Source Available license.
   - You are consuming the Work as a Transitive Package Dependency.
   - You are consuming the Work as a Direct Package Dependency in the capacity of a For-profit company/individual with less than 1M USD annual gross revenue.
   - You are consuming the Work as a Direct Package Dependency in the capacity of a Non-profit organization or Registered Charity.
   ```

Given the license change occured in SixLabors own GitHub repo and isn't immediately apparent from the spectre console github repo, is spectre console aware the change in licensing of the ImageSharp dependency is quite a lot more restrictive than the MIT license spectre console is distributed under?

Currently spectre console is using ImageSharp version 2.1.3, which was released on Jun 18, 2022, see: https://github.com/SixLabors/ImageSharp/releases/tag/v2.1.3 so current usage by spectre console still falls under Apache License 2.0. **The  restrictions introduced by the Six Labors Split License will come into effect if/when a newer version than 2.1.3 of the ImageSharp is included.**

Depending on how FOSS spectre console is intended to be (or not), I also wonder about the possibility of forking the ImageSharp library prior to the Jun 28, 2022 commit to preserve usage under the Apache License (Version 2.0, January 2004) that spectre console was enjoying up until recently. Or alternatively negotiating continuing to license the library under the previous Apache 2.0 license, as is.

**Comment by patriksvensson** [comment]
@FrankRay78 I appreciate your concern, but I think we're covered by the first bullet point: "You are consuming the Work in for use in software licensed under an Open Source or Source Available license.". Apache-2.0 is compatible with MIT. Consumers of Spectre.Console will use ImageSharp as a transitive dependency. They will need to pay for a license if the transitive dependency is not released under an open-source license. Spectre.Console (via Spectre Systems AB), is paying for a license.

@JimBobSquarePants Have I misunderstood something here?

**Reply by JimBobSquarePants** [reply]
@patriksvensson 

As long as Spectre.Console is released under an Open Source or Source Available license then it will continue to consume ImageSharp under the terms of the Apache 2.0 License since the requirements under the Six Labors Split License are met to do so.

This means that nobody downstream that consumes Spectre.Console has to pay anything if they use ImageSharp as a Transitive Package Dependency. 

Should they reference ImageSharp directly as a Direct Package Dependency, i.e., by separately installing a different version from NuGet, then they will have to pay for a Six Labors Commercial License if they do not meet the requirement terms of the Six Labors Split License.

The Split License is explicitly non-viral. I do not want to harm Open Source maintainers. 

@FrankRay78 
>I also wonder about the possibility of forking the ImageSharp library prior to the Jun 28, 2022 commit to preserve usage

I think you may be trivializing the effort that goes into maintaining ImageSharp.

**Reply by FrankRay78** [reply]
> They will need to pay for a license if the transitive dependency is not released under an open-source license. Spectre.Console (via Spectre Systems AB), is paying for a license.

This was also part of my confusion - but from the explanation above, the Apache license granted for the use of ImageSharp within spectre console, is perpetual, irrespective of whether the consumer of spectre console is open sourcing their software or not.

> I think you may be trivializing the effort that goes into maintaining ImageSharp.

Apologies, and not at all @JimBobSquarePants - it's a seriously good library and I appreciate the effort that must go into it.   

Please forgive having to explain the situation to me, it was the following that confused me a little and upon reading the split license I still wasn't 100% sure: 

![image](https://user-images.githubusercontent.com/52075808/202207538-92e303de-bb86-47f3-9d15-a03cd48f5dd5.png)

Perhaps the fact that use of spectre console, commercial or otherwise, carries with it an Apache 2.0 license for ImageSharp can be explained more clearly (at least to luddites like myself) with some wordsmithing of the readme above?  eg. 

_"SixLabors.ImageSharp, a library which Spectre.Console relies upon, is licensed under Apache 2.0 when distributed as part of Spectre.Console. The SixLabors.ImageSharp license covers all other usage, see https://github.com/SixLabors/ImageSharp/blob/master/LICENSE "_

**Reply by patriksvensson** [reply]
@FrankRay78 Absolutely. Would you mind sending a PR with wording you would find clearer? Would appreciate it!

**Reply by FrankRay78** [reply]
FYI - I've now submitted the following PR: https://github.com/spectreconsole/spectre.console/pull/1073   

@JimBobSquarePants - please review the wording (if you get a chance) and suggest any improvements etc.  

I wanted to better explain the usage of ImageSharp and make more explicit the Six Labors Split License for two reasons mainly: 

1. ensure you are getting people buying your commercial license when required, and
2. ensure potential for-profit software developers aren't put off from using spectre.console because they mistakenly believe a SixLabors commercial license is required

**Reply by JimBobSquarePants** [reply]
The language used in the PR is far too complicated. What you wrote above is much better.

>"SixLabors.ImageSharp, a library which Spectre.Console relies upon, is licensed under Apache 2.0 when distributed as part of Spectre.Console. The SixLabors.ImageSharp license covers all other usage, see https://github.com/SixLabors/ImageSharp/blob/master/LICENSE "

**Reply by FrankRay78** [reply]
Thanks @JimBobSquarePants , I'll update the PR accordingly.  I still feel bad about my cack-handed comment on forking the repo, please accept my apology for that.  When I wrote it, I had feared the license change may have inadvertently been pushed out to spectre.console consumers (when upgrading) and I was scratching my head about what could be done.  A poor suggestion and no reflection of my respect for your work.

**Comment by FrankRay78** [comment]
> Spectre.Console (via Spectre Systems AB), is paying for a license

Oh, I really had no idea of this arrangement (and how considerate of Spectre Systems AB)
